_posts/2017-01-05-enhancing-your-webcam.md

```diff
@@ -0,0 +1,102 @@
+That is, the pixel at coordinate (x, y) in the transformed image is copied from the pixel at coordinate f(x, y) in the original image.
+As long as you define f correctly, you can implement swirls, pinches, magnifications, hazes and all sorts of other distortions.
+
+Finally, overlay effects simply overlay new pixels on parts or all of the frame.
+These new pixels can be sourced from anywhere, including other video sources.
+This effectively lets us overlay Giphy videos directly in the camera stream!
+Productivity will never be the same.
+
+Since effects can be chained in Zombocam, the output from one effect’s rendering pass is fed directly as input to the next effect’s rendering pass.
+This opens for a wide array of different possible effect combinations.
+
+
+<img src="/images/zombocam1.png" alt="Zombocam can turn you into a cyclops if you’re not careful when chaining effects!"/>
+<br/>
+
+# Works everwhere! (\*)
+
+In theory, this approach works everywhere out of the box, so you can use when you’re snapping a profile picture on [Facebook](https://facebook.com), hanging out in video meetings on [Appear.in](https://appear.in) or [Google Hangouts](https://hangouts.google.com).
+In practice, however, the story is a little more nuanced.
+Reliably monkey-patching getUserMedia in time in a cross-browser fashion via injection from a WebExtension without going overboard with permissions turns out to be hard in some cases.
+This means that if an application is really adamant at calling getUserMedia reeeally early in the page’s lifetime, getUserMedia might not be monkey-patched yet.
+In that case, Zombocam will simply never trigger, and it will be as if it weren’t ever even installed.
+
+When attempting to transparently monkey-patch APIs one has to take extreme care to make sure that the monkey-patching actually is transparent.
+That means properly forwarding all sorts of properties on the Streams and Tracks returned from getUserMedia that applications might expect and depend on.
+
+One specific example of this that we ran into was with [Appear.in’s new premium offering](https://appear.in/information/premium/), where you can screen-share and show your webcam stream in your meeting room at the same time.
+The application relied on the name of one of the Tracks to be “Screen”, which we didn’t properly forward to our Tracks that we got from our canvas.
+Because of this, Appear.in didn’t know which of the tracks was the screen-sharing track, and things stopped working.
+Properly forwarding the name property solved the issue, and we learned an important lesson in the virtues of actually being transparent when trying to transparently intercept APIs.
+
+# What's next: audio filters
+
+With the new release of [Zombocam](https://www.zombocam.com) we’ve week taken it one step further and enhanced getUserMedia audio tracks as well using the [Web Audio API](https://developer.mozilla.org/en-US/docs/Web/API/Web_Audio_API).
```

Comment:
With the new release of [Zombocam](https://www.zombocam.com) **we’ve week** taken